src/LineBot/RichMenu/RichMenuBounds.cs

```diff
@@ -80,6 +80,9 @@ public int Width
                 if (value < 1)
                     throw new InvalidOperationException("The width cannot be less than 1.");
 
+                if (value + _x > 2500)
+                    throw new InvalidOperationException("The horizontal postion and width will exceed the richMenu's max width.");
```

Comment:
You could use `rich menu` instead of `richMenu` here instead. And we also need this check inside the `X` and `Y` property.